Spigot-API-Patches/0317-Adding-enhanced-Material-Item-Color-Type-API.patch

```diff
@@ -0,0 +1,489 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TheFruxz <cedricspitzer@outlook.de>
+Date: Sun, 13 Jun 2021 15:02:50 +0200
+Subject: [PATCH] Adding enhanced Material/Item Color(Type)-API
+
+
+diff --git a/src/main/java/io/papermc/paper/item/ColorType.java b/src/main/java/io/papermc/paper/item/ColorType.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..752bd3011bdd0aee86c95a84074298aad9fd62c1
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/item/ColorType.java
+@@ -0,0 +1,269 @@
++package io.papermc.paper.item;
++
++import net.kyori.adventure.text.format.TextColor;
++import net.md_5.bungee.api.ChatColor;
++import org.bukkit.Color;
++import org.bukkit.DyeColor;
++import org.bukkit.Material;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.Arrays;
++import java.util.Collections;
++import java.util.Objects;
++import java.util.Set;
++
++/**
++ * Represents different kind of possible Colors, which colorable {@link Material}s also/and {@link DyeableMaterial}s can have
++ */
++public enum ColorType {
++
++    WHITE,
++    ORANGE,
++    MAGENTA,
++    LIGHT_BLUE,
++    YELLOW,
++    LIME,
++    PINK,
++    GRAY,
++    LIGHT_GRAY,
++    CYAN,
++    PURPLE,
++    BLUE,
++    BROWN,
++    GREEN,
++    RED,
++    BLACK;
++
++    /**
++     * Gets all supported colored {@link Material}s, which can be used
++     * @return all possible colored {@link Material}s
++     */
++    @NotNull
++    public static Set<Material> getAllVariants() {
++        Set<Material> result = new java.util.HashSet<>(Collections.emptySet());
```

Comment:
Pre-set size and use static block like Kezz says. Guava ImmutableSet can be used to avoid clones.